pkg/hub/controllers/cluster/reconciler.go

```diff
@@ -50,14 +50,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	if cr.Status.ClusterHealth == nil || time.Since(cr.Status.ClusterHealth.LastUpdated.Time) > 3*time.Minute {
 		log.Info("updating cluster info on controller")
 		if err := r.updateClusterInfo(ctx, cr); err != nil {
+			// Skip the error and continue health check
 			log.Error(err, "unable to update cluster info")
+		} else {
+			return reconcile.Result{Requeue: true}, nil
 		}
 	}
 
 	// Update dashboard creds if it hasn't already
 	if !r.isDashboardCredsUpdated(ctx, cr) {
 		if err := r.updateDashboardCreds(ctx, cr); err != nil {
 			log.Error(err, "unable to update dashboard creds")
+			return reconcile.Result{}, err
+		} else {
+			return reconcile.Result{Requeue: true}, nil
```

Comment:
Why not use RequeueAfter: ReconcileInterval here ?

Reply:
ReconcileInterval is used to run health checks periodically. Here we are updating dashboard creds. We would need to reconcile immedietly to continue health check operations

Reply:
If `updateDashboardCreds` fails why we are not reconciling immedietly ?

Reply:
`return reconcile.Result{}, err` requeues immedietly if error is passed. no need to specify requeue: true